nuxeo-runtime/nuxeo-runtime-metrics/src/main/java/org/nuxeo/runtime/metrics/MetricsDescriptor.java

```diff
@@ -140,16 +160,22 @@ public String prefix() {
         }
 
         protected boolean filter(String name) {
-            if (filters == null || filters.isEmpty()) {
-                String filters = Framework.getProperty(FILTER_PROPERTY, DEFAULT_METRICS);
-                return matchesFilter(name, Arrays.stream(filters.split(",")));
+            if ((whitelistMetrics == null || whitelistMetrics.isEmpty())
+                    && (blacklistMetrics == null || blacklistMetrics.isEmpty())) {
+
+                Stream<String> reportedMetrics = Arrays.stream(
+                        Framework.getProperty(WHITELIST_FILTER_PROPERTY, DEFAULT_WHITELIST_METRICS).split(","));
+                Stream<String> ignoredMetrics = Arrays.stream(
+                        Framework.getProperty(BLACKLIST_FILTER_PROPERTY, DEFAULT_BLACKLIST_METRICS).split(","));
+                return matchesFilter(name, reportedMetrics, ignoredMetrics);
             } else {
-                return matchesFilter(name, filters.stream());
+                return matchesFilter(name, whitelistMetrics.stream(), blacklistMetrics.stream());
             }
         }
 
-        protected boolean matchesFilter(String name, Stream<String> filters) {
-            return filters.anyMatch(f -> ALL_METRICS.equals(f) || name.startsWith(f));
+        protected boolean matchesFilter(String name, Stream<String> reportedMetrics, Stream<String> ignoredMetrics) {
+            return reportedMetrics.anyMatch(f -> ALL_METRICS.equals(f) || name.startsWith(f))
+                    || ignoredMetrics.noneMatch(f -> ALL_METRICS.equals(f) || name.startsWith(f));
```

Comment:
Don't know if we want to handle that, but this condition doesn't handle this case:
Whitelist: `nuxeo.cache`
Blacklist: `nuxeo.cache.user`
In this scenario, we will allow to report `nuxeo.cache.user` metrics whereas it is blacklist.

Reply:
Since I read the extension point documentation, seems to be working like that.